source/appModules/eclipse.py

```diff
@@ -17,7 +26,102 @@ def event_valueChange(self):
 		# Therefore, just drop this event.
 		pass
 
+	def event_caret(self):
+		super(EclipseTextArea, self).event_caret()
+
+		# Check suggestion item and close the list if it is not valid
+		try:
+			if not self.appModule.selectedItem.name:
+				self.event_suggestionsClosed()
+				self.appModule.selectedItem = None
+		except:
+			pass
+
+	@script(
+		gestures = ["kb:enter", "kb:escape"]
+	)
+	def script_closeAutocompleter(self, gesture):
+		gesture.send()
+		
+		# Close the suggestions list if it is opened
+		if self.appModule.selectedItem:
+			self.event_suggestionsClosed()
+			self.appModule.selectedItem = None
+
+	@script(
+		# Translators: Input help mode message for the 'read documentation script
+		description = _("Tries to read documentation for the selected autocompletion item."),
+		gesture = "kb:nvda+d"
+	)
+	def script_readDocumentation(self, gesture):
+		# If there aren't any suggestion selected, there is no way to find quick documentation
+		if not self.appModule.selectedItem:
+			gesture.send()
+			return
+
+		obj = self.appModule.selectedItem.parent.parent.parent.parent.previous.previous

```

Comment:
This seems error prone, what can you do to make this less fragile? Is previous always going to exist? How can we verify that we have the object that we expect?

Reply:
Reworked this completely.

Thanks for your help.